hello-thirdparty/app/build.gradle

```diff
@@ -18,42 +29,34 @@ model {
         }
     }
 
-    compileOptions.with {
-        sourceCompatibility=JavaVersion.VERSION_1_7
-        targetCompatibility=JavaVersion.VERSION_1_7
-    }
-
     android.ndk {
         moduleName = "native-activity"
-        cppFlags.addAll(["-I${file("src/main/jni")}".toString(),
-	                 "-I${gpg_sdk_path}/include".toString()])
+        platformVersion = 9 //same as minSdkVersion.apiLevel for better compatibility
         stl    = "c++_static"
-        ldLibs.addAll(["log", "android", "EGL", "GLESv1_CM", "z"])
-        //
-        //  NOTE: since libgpg is also linked to the stl:
-        //  libc++ explicitly follows it in the list of libraries
-        //  to prevent unresolved stl symbols.
-        //
-        ldLibs.addAll(["gpg", "c++_static"])
+        ldLibs.addAll(["atomic", "log", "android", "EGL", "GLESv1_CM", "z"])
+        abiFilters.addAll(["armeabi", "armeabi-v7a", "x86", "arm64-v8a"]) //filtering ABIs on the ones Google Play Games library is compatible with.
     }
+
+    android.sources {
+        main {
+            jni {
+                dependencies {
+                    library "gpg"
+                }
+            }
+        }
+    }
+
     android.buildTypes {
         release {
             minifyEnabled = false
-            proguardFiles.add(file('proguard-rules.txt'))
-        }
-    }
-    android.productFlavors {
```

Comment:
Flavors? Where we're going, there is no flavors! ;)

ABI-specific flavors were used here only because they were initially required to support native dependencies (and during the Android Dev Summit, someone even said that ABI-specific flavors were "deprecated" now).
If you still prefer to generate one APK per architecture from this sample, there is a blocker: https://code.google.com/p/android/issues/detail?id=192943